client/state/sites/actions.js

```diff
@@ -147,6 +149,13 @@ export function requestSite( siteFragment ) {
 				} );
 			} )
 			.catch( ( error ) => {
+				if (
+					error?.status === 403 &&
+					error?.message === 'API calls to this blog have been disabled.' &&
+					! forceWpcom
+				) {
+					return dispatch( requestSite( siteFragment, true ) );
```

Comment:
At first glance I wondered if this could cause an infinite loop, but I'm assuming that because we're screening for a specific error message that we know indicates a different request path, this is okay.

Reply:
yeah, the `forceWpcom` arg works prevents any further recursion